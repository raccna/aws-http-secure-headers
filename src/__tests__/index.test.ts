import { handler } from '../index';
import { CloudFrontResponseEvent, CloudFrontHeaders } from 'aws-lambda';

const contentTypeHeader = (contentType: string): CloudFrontHeaders => ({
  'content-type': [{ value: contentType }]
});

const fakeRequest = (contentType: string): CloudFrontResponseEvent =>
  ({
    Records: [
      {
        cf: {
          response: {
            headers: contentTypeHeader(contentType),
            status: '200',
            statusDescription: 'OK'
          }
        }
      }
    ]
  } as CloudFrontResponseEvent);

describe('index', () => {
  test('Should process text/html response correctly', async () => {
    const result = await handler(fakeRequest('text/html'));
    expect(result).toMatchSnapshot();
  });

  test('Should process application/json response correctly', async () => {
    const result = await handler(fakeRequest('application/json'));
    expect(result).toMatchSnapshot();
  });
});
