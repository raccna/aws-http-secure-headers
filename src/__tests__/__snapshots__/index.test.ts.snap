// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index Should process application/json response correctly 1`] = `
Object {
  "headers": Object {
    "cache-control": Array [
      Object {
        "key": "Cache-Control",
        "value": "no-cache=\\"Set-Cookie, Set-Cookie2\\"",
      },
    ],
    "content-security-policy": Array [
      Object {
        "key": "Content-Security-Policy",
        "value": "default-src: 'none'",
      },
    ],
    "content-type": Array [
      Object {
        "value": "application/json",
      },
    ],
    "pragma": Array [
      Object {
        "key": "Pragma",
        "value": "no-cache",
      },
    ],
    "referrer-policy": Array [
      Object {
        "key": "Referrer-Policy",
        "value": "same-origin",
      },
    ],
    "server": Array [
      Object {
        "key": "Server",
        "value": "",
      },
    ],
    "strict-transport-security": Array [
      Object {
        "key": "Strict-Transport-Security",
        "value": "max-age=63072000; includeSubdomains",
      },
    ],
    "x-content-type-options": Array [
      Object {
        "key": "X-Content-Type-Options",
        "value": "nosniff",
      },
    ],
    "x-frame-options": Array [
      Object {
        "key": "X-Frame-Options",
        "value": "SAMEORIGIN",
      },
    ],
    "x-xss-protection": Array [
      Object {
        "key": "X-XSS-Protection",
        "value": "1; mode=block",
      },
    ],
  },
  "status": "200",
  "statusDescription": "OK",
}
`;

exports[`index Should process text/html response correctly 1`] = `
Object {
  "headers": Object {
    "cache-control": Array [
      Object {
        "key": "Cache-Control",
        "value": "no-cache=\\"Set-Cookie, Set-Cookie2\\"",
      },
    ],
    "content-security-policy": Array [
      Object {
        "key": "Content-Security-Policy",
        "value": "default-src: 'none'",
      },
    ],
    "content-type": Array [
      Object {
        "value": "text/html",
      },
    ],
    "pragma": Array [
      Object {
        "key": "Pragma",
        "value": "no-cache",
      },
    ],
    "referrer-policy": Array [
      Object {
        "key": "Referrer-Policy",
        "value": "same-origin",
      },
    ],
    "server": Array [
      Object {
        "key": "Server",
        "value": "",
      },
    ],
    "strict-transport-security": Array [
      Object {
        "key": "Strict-Transport-Security",
        "value": "max-age=63072000; includeSubdomains",
      },
    ],
    "x-content-type-options": Array [
      Object {
        "key": "X-Content-Type-Options",
        "value": "nosniff",
      },
    ],
    "x-frame-options": Array [
      Object {
        "key": "X-Frame-Options",
        "value": "SAMEORIGIN",
      },
    ],
    "x-xss-protection": Array [
      Object {
        "key": "X-XSS-Protection",
        "value": "1; mode=block",
      },
    ],
  },
  "status": "200",
  "statusDescription": "OK",
}
`;
